techmatters/terraso-mobile-client
# feat: Design updates to form for Create Project, Edit Project, and Create Site

## Description
Make updates to the form for Create Project, Edit Project, and Create Site

**NOTE:** (Re: Edit Site) I will create an Issue for SiteSettingsScreen to use Formik to make these same updates

### Checklist
- [x] New tests added -- Ohh I'm gonna have to update some test snapshots probably

### Related Issues
Fixes #534 

### Verification steps
1. Click on the Project tab and create a new project.
2. Confirm there is a red "Required" indicator on the name field when it is empty
3. Confirm there is a different red indicator on the name when it has 1 or 2 characters
4. Confirm the name cannot be more than 50 characters
5. Confirm you can have an empty description
6. Confirm the description cannot be more than 72 characters
7. Confirm the Save button is grayed and unpressable when anything is invalid
8. In an existing project, go to Settings. Confirm the same things.
9. Click a location on the homepage map to create a new site. Confirm the same things (minus description ones).
